halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/providers/kubernetes/KubernetesAccount.java

```diff
@@ -47,4 +55,112 @@ public String getKubeconfigFile() {
+    ArrayList<HalconfigProblem> result = new ArrayList<>();
+    io.fabric8.kubernetes.api.model.Config kubeconfig;
+
+    // This indicates if a first pass at the config looks OK. If we don't see any serious problems, we'll do one last check
+    // against the requested kubernetes cluster to ensure that we can run spinnaker.
+    boolean smoketest = true;
+
+    try {
+      File kubeconfigFileOpen = new File(getKubeconfigFile());
+      kubeconfig = KubeConfigUtils.parseConfig(kubeconfigFileOpen);
+    } catch (IOException e) {
+      result.add(new HalconfigProblem(HalconfigProblem.Severity.ERROR, coordinates, e.getMessage()));
+      return result;
+    }
+
+    if (context != null && !context.isEmpty()) {
+      Optional<NamedContext> namedContext = kubeconfig
+          .getContexts()
+          .stream()
+          .filter(c -> c.getName().equals(context))
+          .findFirst();
+
+      if (!namedContext.isPresent()) {
+        result.add(new HalconfigProblem(HalconfigProblem.Severity.ERROR, coordinates,
+            "Context \"" + context + "\" not found in kubeconfig \"" + getKubeconfigFile() + "\"",
+            "Either add this context to your kubeconfig, rely on the default context, or pick another kubeconfig file"));
+        smoketest = false;
+      }
+    } else {
+      String currentContext = kubeconfig.getCurrentContext();
+      if (currentContext.isEmpty()) {
+        result.add(new HalconfigProblem(HalconfigProblem.Severity.ERROR, coordinates,
+            "You have not specified a Kubernetes context, and your kubeconfig \"" + getKubeconfigFile() + "\" has no current-context",
+            "Either specify a context in your halconfig, or set a current-context in your kubeconfig"));
+        smoketest = false;
+      } else {
+        result.add(new HalconfigProblem(HalconfigProblem.Severity.WARNING, coordinates,
+            "You have not specified a Kubernetes context in your halconfig, Spinnaker will use \"" + currentContext + "\" instead",
+            "We recommend explicitly setting a context in your halconfig, to ensure changes to your kubeconfig won't break your deployment"));
+      }
+    }
+
+    if (smoketest) {
+      Config config = KubernetesConfigParser.parse(getKubeconfigFile(), context, cluster, user, namespaces);
+      KubernetesClient client = new DefaultKubernetesClient(config);
+
+      try {
+        client.namespaces().list();
+      } catch (Exception e) {
+        result.add(new HalconfigProblem(HalconfigProblem.Severity.ERROR, coordinates,
+            "Unable to communicate with your Kubernetes cluster: " + e.getMessage(),
+            "Verify that these credentials work manually using \"kubectl\""));
+      }
+    }
+
+    return result;
+  }
+
+  public List<HalconfigProblem> validateDockerRegistries(DeploymentConfiguration deployment, HalconfigCoordinates coordinates) {
```

Comment:
Yes. changes to the docker registries outside this account mean the kubernetes account registries need to be validated without triggering live calls to the kubernetes cluster.